### ISSUE TITLE
How to construct an MCL library that is compatible with WASM

### ISSUE THREAD

**Issue by coolcoolcoo** [herumi/mcl#219]
Pls how do I compile the mcl library to the Wasm version of the library？
Are there any compilation options？

**Comment by herumi** [comment]
Please see https://github.com/herumi/mcl-wasm

**Comment by ndesmoul** [comment]
I have a similar problem, but in my case I have developed a C++ library using MCL that I would like to compile and run in a browser. So just a wasm MCL version that I can call from JS is not what I'm looking for.

Actually I managed to compile my library into wasm, using a specific cmake configuration. MCL initialization seems ok, but as soon as I try to do some computation using MCL it fails. So obviously either I don't use the correct configuration (I think this same configuration worked for older version of MCL though), or some more heavy integration into the JS/NPM ecosystem, like you seem to do with mcl-wasm, is needed. 

What would be the best approach? 
(Preferably one that don't require to rewrite a complete makefile, as my library is relatively complex, and I use cmake...)
